.github/workflows/ci_tests.yml

```
@@ -14,6 +14,90 @@ jobs:
     - name: Use Node.js
       uses: actions/setup-node@v2
       with:
-        node-version: '15'
+        node-version: '14'
 
-    - run: echo "Workflow is added!"
+    - name: Prepare yarn
+      run: |
+        sudo npm i -g yarn@1.22.4 --force
+        yarn --no-progress
+
+    - name: Build
+      run: |
+        yarn build
+        chmod -R 777 .
+
+    - name: Linter & Jest Tests
+      run: |
+        yarn lint
+        yarn test:ci
+
+    - name: Zip artifacts 
+      run: |
+        7z a -tzip -mx3 -mmt2 artifacts.zip packages
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: devextreme-reactive-artifacts
+        path: artifacts.zip
+        retention-days: 1
+
+  test_cafe:
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        component: ['grid', 'chart', 'scheduler']
+    steps:
+    - name: Get sources
+      uses: actions/checkout@v2
+
+    - name: Use Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Prepare yarn
+      run: |
+        sudo npm i -g yarn@1.22.4 --force
+        yarn --no-progress
+
+    - name: Download artifacts
+      uses: actions/download-artifact@v2
+      with:
+        name: devextreme-reactive-artifacts
+
+    - name: Unpack artifacts
+      run: 7z x -y artifacts.zip
+
+    - name: TestCafe Tests Grid
+      run: yarn testcafe:ci:${{ matrix.component }}
+
+  site:
+    runs-on: ubuntu-latest
+    needs: test_cafe
```

Looks like we depend on `build` artifacts, we can change the value of the `needs` field to speed up workflow
```suggestion
    needs: build
```